packages/playwright/src/reporters/teleEmitter.ts

```diff
@@ -236,11 +242,36 @@ export class TeleReporterEmitter implements ReporterV2 {
       duration: result.duration,
       status: result.status,
       errors: result.errors,
-      attachments: this._serializeAttachments(result.attachments),
       annotations: result.annotations?.length ? result.annotations : undefined,
     };
   }
 
+  private _sendNewAttachments(result: reporterTypes.TestResult, testId: string, stepId?: string) {
+    const resultId = (result as any)[this._idSymbol] as string;
+    // Track whether this step (or something else since the last step) has added attachments and send them
+    const knownAttachmentCount = this._resultKnownAttachmentCounts.get(resultId) ?? 0;
+    if (result.attachments.length > knownAttachmentCount) {
+      this._messageSink({
+        method: 'onAttach',
+        params: {
+          testId,
+          resultId,
+          stepId,
+          attachments: result.attachments.slice(knownAttachmentCount).flatMap((attachment: reporterTypes.TestResult['attachments'][number] | undefined) => {
+            // Due to an old bug in users' blobs, the blob may serialize the attachment index as -1,
+            // resulting an undefined attachment. We must preserve the order, but cannot perform normal
+            // serialization on the undefined value
+            if (!attachment)
```

Comment:
I still think we should not care about this case. `attachment` is never undefined. If it is, the blob is busted - oh well.